src/components/inputs/content-switcher/content-switcher.component.tsx

```diff
@@ -9,28 +9,33 @@ import { isTrue } from '../../../utils/boolean-utils';
 import { FormContext } from '../../../form-context';
 import { type FormFieldProps } from '../../../types';
 import FieldValueView from '../../value/view/field-value-view.component';
-import styles from './content-switcher.scss';
+import InlineDate from '../inline-date/inline-date.component';
+import { getQuestionValue } from '../../../utils/common-utils';
 import { useFieldValidationResults } from '../../../hooks/useFieldValidationResults';
 
+import styles from './content-switcher.scss';
+
 const ContentSwitcher: React.FC<FormFieldProps> = ({ question, onChange, handler, previousValue }) => {
   const { t } = useTranslation();
   const [field] = useField(question.id);
   const { setFieldValue, encounterContext, layoutType, workspaceLayout } = React.useContext(FormContext);
   const { errors, setErrors } = useFieldValidationResults(question);
+  const [obsDate, setObsDate] = useState<Date>();
```

Comment:
I'm only worried about making these components aware of the "obs" part of things. It introduces some kind of coupling. @hadijahkyampeire Is it possible to have the inlined date component to subscribe to `question.meta.submission` events in a hook and update the value?

Reply:
@samuelmale I am not sure I understand what you mean, maybe we shall sync on Monday or you can share more insights about the suggestion.